### PR TITLE
Fuel parts auto upgrade

### PR DESCRIPTION
**Pull request by agrrr3**
This is an implementation of auto-upgrading fuel tanks when researching deuterium or antimatter tank technology. 

The implementation adds as fuel as one effect per tech. It reads like:

> Medium Basic Ship Hull:  3
> Ship Part  Basic Tank:     1
>Deuterium Tank Tech:       1
>Antimatter Tank Tech:      3

> Medium Basic Ship Hull..3
> Ship Part  Basic Tank...1
> Deuterium Tank Tech.....1
> Antimatter Tank Tech....3

Also note that there is no check for supply connection. Fuel max gets upgraded instantly when tech is there. But to get fuel, you actually need a refuel source (which in 99% means supply-line connection)

Discussion is in [freeorion forum thread](http://www.freeorion.org/forum/viewtopic.php?f=15&t=10681)

### CODE REVIEW

**Review by dbenage-cx — CHANGES_REQUESTED** [review]
Needs pedia addition(s) and description updates for affected techs.  These techs are still described as unlocking a part, when they instead improve an existing part.

Stringtable entries stemming from the removed ship parts should be removed from the english stringtable.

**Comment by dbenage-cx** [comment]
Antimatter tank needs a new short description, e.g. `SHIP_FUEL_IMPROVE_SHORT_DESC`
Could use it for deuterium tank as well, if the unlock of ramscoop is seen as a side-effect.

No issues noted outside of descriptions on play-through.

**Comment by agrrr3** [comment]
There was no short description previously. Is the key  SHP_ANTIMATTER_TANK_SHORT_DESC and where is it shown?

SHP_ANTIMATTER_TANK_DESC: 
Moderate upgrade to each of your [[shippart FU_BASIC_TANK]] ship parts. Increases [[metertype METER_FUEL]] capacity and thereby the number of jumps your ships can do with refueling.

**Comment by dbenage-cx** [comment]
Referring to the tech definitions https://github.com/freeorion/freeorion/pull/1709/files#diff-2c7aa02d6f884380d6edf227ccf38f06R66
Existing entry points to `SHIP_PART_UNLOCK_SHORT_DESC`, the short description is shown in research window tooltip and at top of pedia entries, e.g. for Deuterium Tank:
deuteriumtanktech

File: default/scripting/techs/ships/Parts.focs.txt
Comment (dbenage-cx) [inline default/scripting/techs/ships/Parts.focs.txt]: At minimum, please document macro variables:
```
// @1@ Accounting label
// @2@ Fuel capacity increase per fuel part
```

File: default/stringtables/en.txt
Comment (dbenage-cx) [inline default/stringtables/en.txt]: Might keep consistent with entry for weapons, e.g. `Improves Ships Fuel`

File: default/stringtables/en.txt
Comment (Cjkjvfnby) [inline default/stringtables/en.txt:12285]: Please remove this and other removed entry from other languages files.

Comment (agrrr3) [reply default/stringtables/en.txt:12285]: We still need the extra fuel tanks. 

Just deuterium and antimatter ship parts are gone. Stringtables are done.

**Comment by agrrr3** [comment]
Looks like I need a rebase?
After that do I have to put up a different pull request?

**Comment by agrrr3** [comment]
English should be ok, german is good.
For all other languages I'm missing a translation for the short description.
SHIP_FUEL_IMPROVE_SHORT_DESC
Improves Tanks Fuel Capacity

Somebody should check what i did for spanish and french.
And can somebody help with the russian translation?

**Comment by dbenage-cx** [comment]
The PR will update with changes to this branch, so a new PR is not necessary (one guide for rebase is https://help.github.com/articles/using-git-rebase-on-the-command-line/ )

Typically with translations, if you are not a native speaker of the language, it is preferred to omit value changes outside of keywords (otherwise it may be a long time for a full review of the PR).

For new entries, if they are excluded from a specific stringtable, it will fall back to the default (english) value.

**Comment by Cjkjvfnby** [comment]
> And can somebody help with the russian translation?

I will fix it after PR merge.

**Comment by agrrr3** [comment]
@dbenage-cx did i address all your issues? The branch is rebased and ready for merge.

**Review by dbenage-cx — APPROVED** [review]

**Comment by dbenage-cx** [comment]
Not sure supported completely for AI, if someone from @freeorion/ai-team can weigh in?
Any concerns for balance @MatGB ?

**Comment by Morlic-fo** [comment]
> Not sure supported completely for AI

Not completely, no. It will not consider the fuel upgrades when making its ship designs.

However, general fuel calculations should only use the maxFuel meter, so this does not hurt the AI. So the lacking AI support for the feature should not hold up the merge, I think.

Regardless, when merging this PR, please open a short issue regarding the lack of AI support and link to this PR so it will not be forgotten.

**Comment by MatGB** [comment]
Cool, it all looks fine from my end.

**Comment by Ouaz** [comment]
> Typically with translations, if you are not a native speaker of the language, it is preferred to omit value changes outside of keywords (otherwise it may be a long time for a full review of the PR).

For new entries, if they are excluded from a specific stringtable, it will fall back to the default (english) value.

+1.

Not a big deal for the FR file (in which your modified translations do not match the english values, for example SHP_DEUTERIUM_TANK_DESC ), as I check it and update it each week. 

Other than this, good work. :)

**Pull request merged**
